core/src/main/java/hudson/Functions.java

```diff
@@ -2299,13 +2299,17 @@ public static String humanReadableByteSize(long size) {
         double number = size;
         if (number >= 1024) {
             number = number / 1024;
-            measure = "KB";
+            measure = "KiB";
```

Comment:
I recommend against making this change and the related changes that correct the previously inaccurate description of the units because users are accustomed to seeing the values displayed in that format and the data entry forms for the node monitor configuration do not accept "KiB", "MiB", or "GiB".  I think it is better for users to have the displayed strings usable as the values in the input forms rather than to have the correct unit of measure displayed but not accepted by the input form.
```suggestion
            measure = "KB";
```

Reply:
New units worked for me as well.  As far as I can tell, the old units are still accepted.  I think that allowing both old and new units in the new implementation is enough.  We don't promise that new settings can be taken backwards to old releases.

The configuration as code settings are especially nice.  It let's me handle some of my static agents as the special cases that they are, while still managing them with configuration as code.  Agents with tiny disc drives will be able to have smaller thresholds while agents with reasonable drives can user larger thresholds for a better margin of safety.  Thanks very much!